fl4health/datasets/skin_cancer/load_data.py

```
@@ -0,0 +1,204 @@
+        image_path = item["img_path"]
+        image = Image.open(image_path).convert("RGB")
+        if transform:
+            image = transform(image)
+        else:
+            # Default transformation if none provided
+            image = transforms.ToTensor()(image)
+        assert isinstance(image, torch.Tensor), f"Image at {image_path} is not a Tensor"
+        target = int(torch.tensor(item["extended_labels"]).argmax().item())
+        return image, target
+
+    with ThreadPoolExecutor(max_workers=num_workers) as executor:
+        results = list(executor.map(load_image, data))
+
+    data_list, targets_list = zip(*results)
+    data_tensor = torch.stack(list(data_list))
+    targets_tensor = torch.tensor(list(targets_list))
+
+    return TensorDataset(data_tensor, targets_tensor)
+
+
+def load_skin_cancer_data(
+    data_dir: Path,
+    dataset_name: str,
+    batch_size: int,
+    sampler: Optional[LabelBasedSampler] = None,
+    train_transform: Union[None, Callable] = None,
+    val_transform: Union[None, Callable] = None,
+    dataset_converter: Optional[DatasetConverter] = None,
+    seed: int = 0,
+) -> Tuple[DataLoader, DataLoader, Dict[str, int]]:
+    """Load skin cancer dataset (training and validation set)."""
+
+    dataset_paths = {
+        "Barcelona": data_dir / "ISIC_2019" / "ISIC_19_Barcelona.json",
+        "Rosendahl": data_dir / "HAM10000" / "HAM_rosendahl.json",
+        "Vienna": data_dir / "HAM10000" / "HAM_vienna.json",
+        "UFES": data_dir / "PAD-UFES-20" / "PAD_UFES_20.json",
+        "Canada": data_dir / "Derm7pt" / "Derm7pt.json",
+    }
+
+    if dataset_name not in dataset_paths:
+        log(ERROR, f"Dataset {dataset_name} not found in available datasets.")
+        raise ValueError(f"Dataset {dataset_name} not found in available datasets.")
+
+    dataset_path = dataset_paths[dataset_name]
+
+    if not dataset_path.exists():
+        log(
+            ERROR,
```

As above, I think we can just raise the exception and they will be logged. Your message here is nice and informative though. So, I'd transfer this message into the exception string.

---

the exception strings now have all these messages